### DISCUSSION TITLE
Unable to change URI of 'To' endpoint

### DISCUSSION THREAD

**Discussion by nbiancolin** [apache/camel-karavan#1263] [Q&A] [answered]
### Ask a question

Title, I have created a route and am trying to add a static endpoint, however I am unable to set the URI of the endpoint.

If I change the URI in the yaml file itself, the object itself also changes depending on the URI (eg. JMS)

### Variant

Vs Code Extension

### Container Management (if applicable)

None

### Operating System (if applicable)

Windows

### Version

4.5.0

**Comment by davsclaus** [comment]
What static endpoint, can you give an example and better show what you do

**Reply by nbiancolin** [reply]
I have been following some camel tutorials and attempting to use Karavan to complete them. In the tutorials they use  `.to`, which I assumed in Karavan meant to use this 'to': 
![image](https://github.com/apache/camel-karavan/assets/117390343/d7252ae7-b827-4e5a-9455-d1d805702065)

when I add it, I am unable to edit the (required) Uri field. 
![image](https://github.com/apache/camel-karavan/assets/117390343/84977e83-c098-402c-a12a-8b0e0dd56880)

I can edit it in the Yaml file and it appears* to work, but I am not sure if that is the expected behaviour

**Reply by davsclaus** [reply]
Ah yeah @mgubaidullin maybe users should be able to type in the uri field for this `to` EIP in this situation.

**Reply by davsclaus** [reply]
@nbiancolin you can use `toD` that allows to edit the to field where you can type in the static url

**Reply by mgubaidullin** [reply, marked as answer]
@nbiancolin In Karavan you do not need to use `to` explicitly , just select a Component you need to connect with and Karavan will set `to` for you

**Reply by nbiancolin** [reply]
Thank you for the help! Just realized now I was digging through the Integration Patterns menu instead of the Components menu and managed to get it sorted.
